src/App.tsx

```diff
@@ -14,11 +14,134 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const numbers: number[] = Array.from(Array(10).keys());
+
+type State = {
+  goods: string[];
+  goodLength: string;
+  isVisible: boolean
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    goodLength: '1',
+    isVisible: false,
+  };
+
+  reversedGoods = () => {
+    this.setState((state) => ({
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortedGoods = () => {
+    this.setState((state) => ({
+      goods: [...state.goods].sort((a, b) => a.localeCompare(b)),
+    }));
+  };
+
+  resetGoods = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+      goodLength: '1',
+    });
+  };
+
+  sortByLengthGoods = () => {
+    this.setState((state) => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  changeLength = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const newGoods = goodsFromServer.filter(
+      good => good.length >= Number(event.target.value),
+    );
+
+    this.setState({
+      goodLength: event.target.value,
+      goods: newGoods,
+    });
+  };
+
+  showGoods = () => {
+    this.setState({ isVisible: true });
+  };
+
+  hideGoods = () => {
+    this.setState({ isVisible: false });
+  };
```

Comment:
can you combine showGoods and hideGoods?

Reply:
remove)